### PR TITLE
doc: Making examples visible to users

### LINKED ISSUE #2819: Organize Dafny tutorial examples
Organize various tutorial examples, such as the ones under Test/tutorial, for user consumption, in docs.
Implement a means so that such examples are tested as part of the regular CI testing.

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #2819

Make examples visible to users, while still including them in CI

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/examples/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: Miscellaneous Dafny Examples
+---
+
+The links below connect to annotated examples of Dafny source code,
+demonstrating various coding techniques or proof capabilities..
```
Comment (robin-aws) [inline docs/examples/README.md:6]:
```suggestion
demonstrating various coding techniques or proof capabilities.
```

File: docs/examples/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Each of these prove successfully.
```
Comment (robin-aws) [inline docs/examples/README.md:7]:
```suggestion
Each of these verify successfully.
```

File: docs/examples/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+The files for these examples can be found in 
+the [examples folder of Dafny's github project](https://github.com/dafny-lang/dafny/tree/master/docs/examples).
+
+- [Simple Maximum method](https://github.com/dafny-lang/dafny/tree/master/Test/examples/maximum.dfy)
+- [Tutorial on parser combinators](https://github.com/dafny-lang/dafny/tree/master/Test/examples/parser_combinators.dfy)
+- A simple compiler: [code](https://github.com/dafny-lang/dafny/tree/master/Test/examples/Simple_compiler/Compiler.dfy) and [discussion](../../Test/examples/Simple_compiler/README)
```
Comment (robin-aws) [inline docs/examples/README.md:14]: Consider just linking to the README, since it provides the context and names `Compiler.dfy` itself anyway.

**Review by robin-aws — COMMENTED** [review]
I like the simple solution, and I agree with relabelling them as examples rather than tutorials.

File: docs/examples/README.md (hunk 3 above)
Comment (davidcok) [reply docs/examples/README.md]: Done

**Review by robin-aws — APPROVED** [review]

**Pull request merged**